elide-datastore/elide-datastore-inmemorydb/src/main/java/com/yahoo/elide/datastores/inmemory/InMemoryTransaction.java

```diff
@@ -186,4 +187,36 @@ public void close() throws IOException {
         ConcurrentHashMap<String, Object> objs = dataStore.get(entityClass);
         return objs == null ? 0L : objs.size();
     }
+
+    /**
+     * Visitor to retrieve a properly combined in-memory filter.
+     */
+    public class InMemoryFilter implements Visitor<java.util.function.Predicate> {
```

Comment:
There are already two visitors which do something very similar.  Look at InMemoryFilterVisitor and to a lesser extent FilterExpressionCheckEvaluationVisitor